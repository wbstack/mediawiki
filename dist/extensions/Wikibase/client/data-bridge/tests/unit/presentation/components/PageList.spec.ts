import { shallowMount } from '@vue/test-utils';
import MediaWikiRouter from '@/definitions/MediaWikiRouter';
import PageList from '@/presentation/components/PageList.vue';

describe( 'PageList', () => {
	const router: MediaWikiRouter = {
		getPageUrl: jest.fn().mockImplementation( ( title ) => `https://wiki/${title}` ),
	};
	const pages = [ 'Foo', 'Bar', 'Baz' ];

	it( 'builds a list matching our snapshot', () => {
		const wrapper = shallowMount( PageList, {
			propsData: {
				router,
				pages,
			},
		} );
		expect( wrapper.element ).toMatchSnapshot();
	} );

	it( 'calls the router to build the link for each page', () => {
		shallowMount( PageList, {
			propsData: {
				router,
				pages,
			},
		} );
		expect( router.getPageUrl ).toHaveBeenCalledTimes( pages.length );
		expect( router.getPageUrl ).toHaveBeenNthCalledWith( 1, pages[ 0 ] );
		expect( router.getPageUrl ).toHaveBeenNthCalledWith( 2, pages[ 1 ] );
		expect( router.getPageUrl ).toHaveBeenNthCalledWith( 3, pages[ 2 ] );
	} );
} );
