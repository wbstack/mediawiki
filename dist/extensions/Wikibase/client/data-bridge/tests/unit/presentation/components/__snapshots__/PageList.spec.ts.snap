// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageList builds a list matching our snapshot 1`] = `
<ul>
  
  <li>
    <a
      href="https://wiki/Foo"
    >
      Foo
    </a>
  </li>
  <li>
    <a
      href="https://wiki/Bar"
    >
      Bar
    </a>
  </li>
  <li>
    <a
      href="https://wiki/Baz"
    >
      Baz
    </a>
  </li>
  
</ul>
`;
